python/lsst/ctrl/oods/scanner.py

```diff
@@ -0,0 +1,34 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import asyncio
+import os
+
+
+class Scanner(object):
+
+    async def scan(self, directory):
+        await asyncio.sleep(0)
+        for entry in os.scandir(directory):
+            await asyncio.sleep(0)
+            yield entry
+            if entry.is_dir(follow_symlinks=False):
+                async for sub_entry in self.scan(entry.path):
+                    yield sub_entry
```

Comment:
I'm far from being an `asyncio` expert so I might be wrong, but it looks like the lines 33-34 could be replaced with
```python
yield from self.scan(entry.path)
```

Reply:
From what I read, the "yield from" is considered the "old" way of doing this, with the method above preferred now.